### PR TITLE
Only run Naming Rules on "Ordinary" methods

### LINKED ISSUE #9710: [PR Available] Naming Style: Analyzer treats constructor name as .ctor
**Steps to Reproduce**:

1. Create a rule that all methods should be PascalCased, and set it as an error
2. Write a class with a constructor

**Expected Behavior**:
Ignore constructors, since their naming rule should come from the type.

**Actual Behavior**:
Violation flagged, and fix suggests renaming `.ctor`.

### LINKED ISSUE #9711: [PR Available] Naming style: Property accessors shouldn't be analyzed
**Steps to Reproduce**:

1. Create a rule that Methods should be PascalCased and make violations an error
2. Write a property

**Expected Behavior**:
No violations.

**Actual Behavior**:
`get` and `set` are squiggled, and the fix suggests to rename `get_Property` or `set_Property`.

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes #9710, Fixes #9711

### CODE REVIEW

File: src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStylePreferencesInfo.cs [hunk 1]
```diff
@@ -33,5 +40,16 @@ internal bool TryGetApplicableRule(ISymbol symbol, ISymbolCategorizationService
             applicableRule = matchingRule.GetBestMatchingRule(symbol, categorizationService);
             return true;
         }
+
+        private bool IsSymbolNameAnalyzable(ISymbol symbol)
+        {
+            var methodSymbol = symbol as IMethodSymbol;
+            if (methodSymbol != null && methodSymbol.MethodKind != MethodKind.Ordinary)
```
Comment (Pilchie) [inline src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStylePreferencesInfo.cs]: Are there things that should not be analyzable for things that are not methods?  Explicitly implemented properties are the only ones that come to mind quickly.

**Comment by Pilchie** [comment]
:+1: when tests appear.

**Comment by jaredpar** [comment]
test determinism please

**Comment by jaredpar** [comment]
That last comment was to verify my Jenkins change has taken effect.  It appears to have.  Don't need to wait on that job to fintish unless you'd like to.

**Comment by dpoeschl** [comment]
@dotnet/roslyn-ide Ping for review

File: src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+            var info = new SerializableNamingStylePreferencesInfo();
+            info.SymbolSpecifications.Add(symbolSpecification);
+            info.NamingStyles.Add(namingStyle);
+            info.NamingRules.Add(namingRule);
+
+            return info.CreateXElement().ToString();
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs]: This isn't a `WpfFact` anymore.

File: src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            Options(new OptionKey(SimplificationOptions.NamingPreferences, LanguageNames.CSharp), MethodNamesArePascalCaseOptionString());
+
+        private IDictionary<OptionKey, object> Options(OptionKey option, object value)
+        {
+            var options = new Dictionary<OptionKey, object>();
+            options.Add(option, value);
+            return options;
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.NamingStyle)]
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs]: This isn't `WpfFact` anymore.

File: src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            var symbolSpecification = new SymbolSpecification(
+                Guid.NewGuid(),
+                "Name",
+                SpecializedCollections.SingletonEnumerable(new SymbolSpecification.SymbolKindOrTypeKind(SymbolKind.Method)).ToList(),
+                SpecializedCollections.EmptyList<SymbolSpecification.AccessibilityKind>(),
+                SpecializedCollections.EmptyList<SymbolSpecification.ModifierKind>(),
+                SpecializedCollections.EmptyList<string>());
+
+            var namingStyle = new NamingStyle();
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs]: You could use an object initializer here.

File: src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            namingStyle.CapitalizationScheme = Capitalization.PascalCase;
+            namingStyle.Name = "Name";
+            namingStyle.Prefix = "";
+            namingStyle.Suffix = "";
+            namingStyle.WordSeparator = "";
+
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs]: Extra blank line.

File: src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStylePreferencesInfo.cs (hunk 1 above)
Comment (jasonmalinowski) [inline src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStylePreferencesInfo.cs]: Use ?. here? It makes the method much smaller.

File: src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs (hunk 5 above)
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs]: Extra blank line.

File: src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs]: Extra blank line.

**Comment by Pilchie** [comment]
:+1:

File: src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                Guid.NewGuid(), 
+                "Name", 
+                SpecializedCollections.SingletonEnumerable(new SymbolSpecification.SymbolKindOrTypeKind(TypeKind.Class)).ToList(),
+                SpecializedCollections.EmptyList<SymbolSpecification.AccessibilityKind>(),
+                SpecializedCollections.EmptyList<SymbolSpecification.ModifierKind>(),
+                SpecializedCollections.EmptyList<string>());
+
+            var namingStyle = new NamingStyle();
+            namingStyle.CapitalizationScheme = Capitalization.PascalCase;
+            namingStyle.Name = "Name";
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/NamingStyles/NamingStylesTests_OptionSets.cs]: Localized?

File: src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStylePreferencesInfo.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStylePreferencesInfo.cs]: +1 4 ?.

**Comment by Pilchie** [comment]
Retesting since results are a little old, but let's try to get this merged.

@dotnet-bot retest this please.

**Comment by rchande** [comment]
retest this please

**Comment by rchande** [comment]
@dotnet-bot retest eta please

**Comment by rchande** [comment]
Failure was inline peek, seems unrelated.

**Pull request merged**